finite_state_machine/state_machine.py

```diff
@@ -12,6 +13,24 @@ def __init__(self):
         except AttributeError:
             raise ValueError("Need to set a state instance variable")
 
+    def all_transitions(self):
+        class_fns = inspect.getmembers(self, predicate=inspect.ismethod)
+        transitions = []
+        for func in class_fns:
+            if hasattr(func[1], "__fsm"):
+                meta = getattr(func[1], "__fsm")
+                for val in list(meta.transitions.values()):
+                    transitions.append(val)
+        return transitions
+
+    def available_transitions(self):
+        all_transitions = self.all_transitions()
+        avail_transitions = []
+        for transition in all_transitions:
+            if transition.source == self.state:
+                avail_transitions.append(transition)
+        return avail_transitions
```

Comment:
This shouldn't change while the program is running; you can probably just create as it's being read and then do a lookup to get the data as required.

Reply:
Hello Aly, where do you think this should be stored? Originally, I wanted to create a table and save it as a single attribute to the SateMachine - however, I ran into issues accessing the StateMachine object from the `transition_decorator` function (probably stemming from my lack of experience with decorators).

Reply:
You can access the machine class inside the decorator. This is where it is being extracted: https://github.com/alysivji/finite-state-machine/blob/bb96cf33489397eccaa21d906d7a7e64954150d1/finite_state_machine/state_machine.py#L52-L55

Think the state machine should be called `machine` instead of `self`.

You can add a `.__fsm` attribute to the class which contains information you need. Might make sense to use a namedtuple like how its done for functions, depending on how complex the meta information you want to store is.

Create a basic decorator and play around with it. Inspect it in the debugger (or print to the console) so you understand how the control flow works.

Also recommend looking at `collections.defaultdict`

Reply:
From my testing yesterday, I was thinking that line 52-55 above only runs when you call a child class method with a transition decorator. Meaning you could only write data to the .__fsm attribute when you called a method vs. on init of the child class. I will keep working with it, as it will give me something to do this weekend haha.

Reply:
Ahhh, good point. I can take a look at this tomorrow. I have a few things on my todo list.

Not quite sure what the best implementation is. Might have to use the inspect module before calling the wrapper function.

Reply:
Taking a looking at the [docs](https://docs.python.org/3/library/inspect.html) looks like you can access it via `function.__self__`.

Just did a test in IPython. It pulls the object instance versus class.

Reply:
That is where I got stuck in my original approach, and I defaulted to how django-fsm did it. Appreciate your patience and help Aly! I will try and correct some things this weekend to get it closer.